packages/keystone/KSv5v6/v5/prepareKeystone.js

```diff
@@ -241,6 +247,44 @@ process.on('unhandledRejection', (err, promise) => {
     }
 })
 
+process.on('SIGPIPE', async () => {
+    if (process.env['DISABLE_HEAP_SNAPSHOT']) {
+        console.log('Dumping of heap snapshots disabled')
+        return
+    }
+
+    try {
+        const labelCreateSnapshot = 'Heap snapshot created in'
+
+        console.log('Start dumping heap snapshot')
+        console.time(labelCreateSnapshot)
+        const fileName = v8.writeHeapSnapshot()
+        console.timeEnd(labelCreateSnapshot)
+        console.log(`Created snapshot file: ${fileName}`)
+
+        const labelUploadSnapshot = 'Snapshot uploaded in'
+
+        console.log('Start snapshot uploading')
+        console.time(labelUploadSnapshot)
+        const ExportFileAdapter = new FileAdapter('heapSnapshots')
+        const stream = fs.createReadStream(path.resolve(fileName))
+        const fileInfo = await ExportFileAdapter.save({
+            stream,
+            filename: fileName,
+            id: `snapshot-${fileName}`,
+        })
+        console.timeEnd(labelUploadSnapshot)
+
+        const { filename } = fileInfo
+        const url = ExportFileAdapter.publicUrl({ filename })
+
+        console.log(`Snapshot file info: ${JSON.stringify(fileInfo)}`)
+        console.log(`Snapshot file URL: ${url}`)
+    } catch (err) {
+        console.error(err)
+    }
```

Comment:
Should it also fire `process.exit` at the end of that callback?

Reply:
No, we do not need to kill process. try-catch is just for preventing process failing.

Reply:
This callback work only if someone execute `kill -SIGPIPE <pid>` in console. Node process will not be killed.